lib/dash/content_protection.js

```diff
@@ -262,6 +262,169 @@ shaka.dash.ContentProtection.parseFromRepresentation = function(
 };
 
 
+/**
+ * Gets a Widevine license URL from a content protection element
+ * containing a custom `ms:laurl` element
+ *
+ * @param {shaka.dash.ContentProtection.Element} element
+ * @return {string}
+ */
+shaka.dash.ContentProtection.getWidevineLicenseUrl = function(element) {
+  const mslaurlNode = shaka.util.XmlUtils.findChild(element.node, 'ms:laurl');
+  if (mslaurlNode) {
+    return mslaurlNode.getAttribute('licenseUrl') || '';
+  }
+  return '';
+};
+
+
+/**
+* @typedef {{
+*   type: number,
+*   value: string
+* }}
+*
+* @description
+* The parsed result of a PlayReady object record.
+*
+* @property {number} type
+*   Type of data stored in the record.
+* @property {string} value
+*   Record content.
+*/
+shaka.dash.ContentProtection.PlayReadyRecord;
+
+/**
+ * Enum for PlayReady record types.
+ * @enum {number}
+ */
+shaka.dash.ContentProtection.PLAYREADY_RECORD_TYPES = {
+  RIGHTS_MANAGEMENT: 0x001,
+  RESERVED: 0x002,
+  EMBEDDED_LICENSE: 0x003,
+};
+
+
+/**
+* Parses a Uint16Array for PlayReady Object Records.
+* Each PRO Record is preceeded by it's PlayReady
+* Record type and length in bytes.
+*
+* PlayReady Object Record format: https://goo.gl/FTcu46
+*
+* @param {Uint16Array} recordData
+* @return {!Array.<shaka.dash.ContentProtection.PlayReadyRecord>}
+* @private
+*/
+shaka.dash.ContentProtection.parseMsProRecords_ = function(
+  recordData) {
+  let head = 0;
+  let records = [];
+
+  while (head < recordData.length) {
+    const recordRaw = recordData.subarray(head);
+    if (recordRaw.length > 1) {
+      let offset = 0;
+      const type = recordRaw[offset++];
+      const byteLength = recordRaw[offset++];
+      // Convert from bytes to WORD length
+      const wordLength = byteLength / 2;
+      const end = wordLength + offset;
+      // subarray end is exclusive
+      const recordValue = recordRaw.subarray(offset, end);
+      const recordString = shaka.util.StringUtils.fromUTF8(recordValue);
+      records.push({
+        type: type,
+        value: recordString,
+      });
+      head = end;
+    }
+  }
+
+  return records;
+};
+
+
+/**
+* PlayReady Object format: https://goo.gl/W8yAN4
```

Comment:
While I like having the link for "more" information, the core information needs to be here too. The resource that the link points to can go away and then we lose the information we need.

Reply:
Added more info to doc string.